video/audio_channel.h

```diff
@@ -373,11 +399,19 @@ bool AudioChannel::isFinished(uint32_t elapsed) {
 }
 
 void AudioChannel::loop() {
+	// _mutex.lock();
+	if (_mutex.try_lock() == false) {
```

Comment:
our loop function is running ever 1ms, and is responsible for driving the channel state machine

if we can't get a lock on our channel mutex then stuff is changing that may negatively affect our ability to move to the next state-machine step.  we therefore will drop out and wait for the next loop

Reply:
mutex lock here has now been removed

this is essentially safe, since state adjustments from outside of the loop thread only trigger new states within the state machine when it's safe to do so.

the use of locking was primarily related to adding/removing waveform generators to the underlying sound system, which is, essentially, already done in a reasonably safe manner